code/dlgo/chess/chessboard.py

```diff
@@ -0,0 +1,36 @@
+import chess as _chess
+from dlgo.gotypes import Player
+
+__all__ = [
+    'GameState',
+]
+
+class GameState:
+    def __init__(self, board):
+        self.board = board
+        self.next_player = Player.white if board.turn else Player.black
+
+    def apply_move(self, move):
+        """Return the new GameState after applying the move."""
+        next_board = self.board.copy()
+        next_board.push(move)
+        return GameState(next_board)
+
+    @classmethod
+    def new_game(cls):
+        board = _chess.Board()
```

Comment:
I'm a bit confused about your `_chess` import. where do I find the code for `Board`, for instance?

Reply:
That's the `python-chess` part. The code for board is here: https://python-chess.readthedocs.io/en/latest/core.html#chess.Board